### ISSUE TITLE
Cannot build with Java 22

### ISSUE THREAD

**Issue by RyanTheAllmighty** [ATLauncher/ATLauncher#886]
### What happened?

Cannot build the project with Java 22 due to Gradle being outdated. Sounds like we're waiting on a Groovy update as well:

> Gradle now supports [using Java 22](https://docs.gradle.org/8.7/userguide/compatibility.html#java) for compiling, testing, and starting other Java programs. Selecting a language version is done using [toolchains](https://docs.gradle.org/8.7/userguide/toolchains.html).
> You cannot run Gradle 8.7 itself with Java 22 because Groovy still needs to support JDK 22. However, future versions are expected to provide this support.

There may be some work to upgrade as per https://docs.gradle.org/8.7/userguide/upgrading_version_8.html#changes_8.3

### Operating System

Windows

### Is this actually a launcher bug?

- [X] I am sure this is a bug with the launcher and not an issue with running Minecraft

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RyanTheAllmighty** [comment]
I don't think this can be closed yet. From https://docs.gradle.org/8.7/release-notes.html#support-for-building-projects-with-java-22:~:text=and%20usability%20improvements-,Support%20for%20building%20projects%20with%20Java%2022,-Gradle%20now%20supports

> You cannot run Gradle 8.7 itself with Java 22 because Groovy still needs to support JDK 22. However, future versions are expected to provide this support.

This is the main issue we have currently with running AUR builds has been this and is currently being worked around in a hacky way

**Comment by Doomsdayrs** [comment]
> I don't think this can be closed yet. From https://docs.gradle.org/8.7/release-notes.html#support-for-building-projects-with-java-22:~:text=and%20usability%20improvements-,Support%20for%20building%20projects%20with%20Java%2022,-Gradle%20now%20supports
> 
> > You cannot run Gradle 8.7 itself with Java 22 because Groovy still needs to support JDK 22. However, future versions are expected to provide this support.
> 
> This is the main issue we have currently with running AUR builds has been this and is currently being worked around in a hacky way.
> 
> Or is the fact we moved to Kotlin solve this since it only mentions Groovy. Can it just be tested with Java 22 first?

I tested it with Java 22 on my system, thus closing it as I found no issues.

Did you do a build test yet?

**Comment by RyanTheAllmighty** [comment]
Nah just wanted to make sure it was all good

**Comment by RyanTheAllmighty** [comment]
Fixed with d74997d04fb8c8caf3d5a69215d2f1120ca5c697
